### ISSUE TITLE
Remove Application Registry and Connector Service from Kyma

### ISSUE THREAD

**Issue by janmedrek** [kyma-project/kyma#13718]
**Description**

Application Registry and Connector Service should be removed from Application Connector. There should be Compass and non-Compass versions of the Application Connector chart, it should also be simplified as much as possible.

In short, the default Kyma OS installation should no longer contain:
- Application Registry
- Connector Service
- Connection Token Handler
- Connectivity Validator (still present in the Compass/managed scenario)
- Istio mTLS gateway (still present in the Compass/managed scenario)
- Various jobs that set up the certificate flow (may still be needed in Compass/managed scenario)

The **default flow with Compass stays unchanged**, so we need to make sure that all the relevant components stay, thus **it should be possible to configure the whole Application Connector chart to either Compass or non-Compass mode**.

This change will force either:
- A simplified setup, where certificates are not needed in order to send events and the registration flow is replaced by a single YAML file containing Application CR (will be described in the latter part as a blogpost explaining that setup).
- A setup where Kyma and Compass are talking to each other and are covering the existing functionalities of certificate setup and registration flow.

The pipelines testing the flow should be obviously removed, **we need to keep those that do the Compass integration** as it will be the new way of working with Kyma and certificates.

**Reasons**

We are maintaining the solution for a very long time already. Since it was [deprecated back in mid-2020](https://kyma-project.io/blog/2020/5/29/release-notes-113#deprecated-connector-and-application-registry-ap-is) it brought near-zero value and only resulted in big maintenance effort regarding pipelines and security issues. We do have alternatives in form of Compass or other approaches to Kyma-External System communication. That's why we've decided to drop the flow and rely completely on Compass in this area.

This change will reduce the maintenance effort, simplify the whole design, and drop one of the more-complicated flows that are not really needed anymore in Kyma.

**Acceptance Criteria**
- Application Connector available in Compass and non-Compass version
- Non-Compass flavor does not support certificate and registration flows
- Flow with Compass stays untouched
- Unnecessary components and jobs removed
- Tests and pipelines adjusted and added/removed
- Whole Compass-related flow tested with an external system

**Comment by janmedrek** [comment]
Here is a short visual representation of what needs to be removed. The components marked in red are to be simply deleted, while the orange ones have some use-case in the Compass scenario, and should only be present if Compass Runtime Agent and its configuration has been applied on the cluster.

![app-connector drawio](https://user-images.githubusercontent.com/22869996/160391949-4f051916-f75a-48ef-88b9-f07554bdd6d0.png)

And this is what is required by the Compass flow:
![app-connector-compass drawio](https://user-images.githubusercontent.com/22869996/160391964-01412dbf-4fc6-417a-bc38-66d6dc4fe85a.png)

**Comment by Disper** [comment]
@janmedrek  What about the current registration flow in open source and security of receiving events? Does that mean that we will not secure the events endpoints that was previously secured via mTLS?

**Comment by Disper** [comment]
Work plan
Part 1 - (2)
Remove pipelines and sources (start with pipelines)
- [ ] Application Registry
- [ ] Connector Service
- [ ] Connection Token Handler

Part 2 - (5)
Modify Fast integration tests for the OS scenario. There is no need to modify CommerceMock. 
- [ ] Remove certificate generation and api generation from the flow. Instead we create application CRD that contains APIs so Central Application Gateway can proxy requests. 

Part 3 - (2)
Remove subcharts responsible for installation and , verify that it works
- [ ] Application Registry
- [ ] Connector Service
- [ ] Connection Token Handler
- [ ] install chart using both ways to test it

Part 4 documentation - (8)
- [ ] update tutorials and readmes are adapted accordingly, including diagrams

**Comment by janmedrek** [comment]
**Part 5 - provide a clear description of OS alternatives**
- Prepare a blog post containing:
  - a way to expose internal eventing
  - a guide on how to prepare Application CR for external systems (i.e. C4C/CC)
  - kyma + compass integration guide (?)

**Comment by janmedrek** [comment]
> Remove certificate generation and api generation from the flow. Instead we create application CRD that contains APIs so Central Application Gateway can proxy requests.

Regarding that: 
If we want to preserve the flow then we also need to find a way to send an event to Kyma, without modifying the Commerce Mock it won't be possible, right? The other option would be to send an event manually to an exposed API Rule, but what is the point of using the Commerce Mock in this case then?

The easier solution would be to cover it with the Compass Scenario, as it tests the basic flow for Kyma: mTLS, eventing, Application Gateway, and does not require introducing any changes in the Commerce Mock. It also makes sense as we will not be able to introduce such changes in any External System integrating with Kyma.

**Comment by Disper** [comment]
@kyma-project/framefrog could we remove/rewrite the tests so we will unblock the pipelines first?

**Comment by koala7659** [comment]
**Part3 workplan**

- [ ] Remove connector-service, app-registry, connection-token-handler
- [ ] Remove cert setup jobs
- [ ] Adjust installation for connector gateway installation on SKR

**Comment by mvshao** [comment]
### Part 1 workplan
**Remove pipelines:**
 - [ ] Application Registry
 - [ ] Connector Service
 - [ ] Connection Token Handler
 
 **Remove sources:**
 - [ ] Application Registry
 - [ ] Connector Service
 - [ ] Connection Token Handler

**Comment by koala7659** [comment]
@mvshao Please remember we are removing  also connectivity validator in KymaOS scenario, please refer to the diagram in the issue for details

**Comment by franpog859** [comment]
>We (comrade @franpog859 please help me here) would need to check what is covered by the "Get Started" test flow, if in-cluster eventing is included then we are good to go with dropping the other flow at all. So in the end eventing functionality will be tested by exposing the in-cluster eventing with the APIRule and calling that.

@janmedrek , the fast-integration open-source getting-started-guide flow looks as follows:
1. [It creates a microservice with some Redis, ApiRule, ServiceInstance, Binding, and a BindingUsage](https://github.com/kyma-project/kyma/blob/main/tests/fast-integration/test/fixtures/getting-started-guide/index.js#L121-L135)
1. Calls directly the microservice via the VirtualService to [add something to the Redis](https://github.com/kyma-project/kyma/blob/main/tests/fast-integration/test/fixtures/getting-started-guide/index.js#L101-L109) and later [read from it](https://github.com/kyma-project/kyma/blob/main/tests/fast-integration/test/fixtures/getting-started-guide/index.js#L90-L99)

**Comment by franpog859** [comment]
The test decision (12.04.2022): remove the Commerce Mock flow from the non-Compass pipeline and consider enhancing the SKR pipeline with the Eventing test

**Comment by majakurcius** [comment]
So far we've supported registering APIs in Application Registry. 
Do we not support it anymore or is there an alternative way for it? 

Depending on the answer, either [this whole section in our documentation](https://kyma-project.io/docs/kyma/latest/01-overview/main-areas/application-connectivity/ac-01-application-connector/#supported-ap-is) must be removed, or just the [link](https://kyma-project.io/docs/kyma/latest/04-operation-guides/operations/ac-02-api-registration/) in its first sentence must be removed/replaced.

**Comment by majakurcius** [comment]
TODO doc-wise: 

- [ ] Remove all occurrences of `Application Registry` and `Connector Service` in the documentation on the Website
- [ ] Go through the documentation and remove the information that is no longer true after the removal
- [ ] Remove all occurrences of `Application Registry` and `Connector Service` in READMEs
- [ ] Update numbering in doc names
- [ ] Update links, if needed

**Comment by majakurcius** [comment]
@janmedrek 
Please confirm that this change is correct: 

**Application Connector Architecture**

BEFORE
![image](https://user-images.githubusercontent.com/53601578/164995968-cb48d519-2850-4677-8e29-f776b7125670.png)

AFTER
![image](https://user-images.githubusercontent.com/53601578/164996792-32eeeb5e-aee1-4668-a17d-386528c646ca.png)

**Comment by majakurcius** [comment]
Doc-wise: what's left is to adjust the remaining tutorials to the new flow. 
I'll be working with @akgalwas on this.

**Comment by mvshao** [comment]
What is left to do:
- adjust the Fast Integration Tests and make sure that the new flow is operating well- @VOID404 and @koala7659 are working on it
- merge the changes from PRs, regarding the removal of the pipelines, sources, helm charts and Fast Integration Tests

**Comment by majakurcius** [comment]
The READMEs have been checked and dealt with.
Adjusting the tutorials is work in progress.

**Comment by akgalwas** [comment]
We need to remove application-connector-tests as it is not used in pipelines and references Application Registry.

**Comment by majakurcius** [comment]
For documentation progress tracking, refer to the [comment above](https://github.com/kyma-project/kyma/issues/13718#issuecomment-1100143376).

**Comment by koala7659** [comment]
New PR for charts with removed components https://github.com/kyma-project/kyma/pull/14225

**Comment by koala7659** [comment]
Where we are now: 

- [x] Fast integration tests fixed and reviewed
- [x] All identified pipelines pass
- [x] Chart update + FIT update merged to common PR #14225 
- [ ] Approval for tracing tests changes
- [ ] Approval for eventing tests changes
- [ ] Merge to main [PR #5232](https://github.com/kyma-project/test-infra/pull/5232) - pipelines
- [ ] Merge to main PR #14225 
- [ ] Documentation approval
- [ ] Merge to main PR #13891  
- [ ] Merge documentation

**Comment by franpog859** [comment]
It's already on main, @Sawthis . We are double-checking if everything works fine on the DEV environment

**Comment by arturskorupa** [comment]
Awaiting migration script to be tested on DEV (PR https://github.com/kyma-project/kyma/pull/14413)

**Comment by mareknawa** [comment]
@akgalwas what is the current status with the script? The issue is release blocker and we need it fixed ASAP if we want to release today.
